sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerWriteSchemaTransformProvider.java

```diff
@@ -183,20 +262,48 @@ public abstract static class SpannerWriteSchemaTransformConfiguration implements
     @SchemaFieldDescription("Specifies the Cloud Spanner table.")
     public abstract String getTableId();
 
+    @SchemaFieldDescription("Specifies how to handle errors.")
+    @Nullable
+    public abstract ErrorHandling getErrorHandling();
+
     public static Builder builder() {
       return new AutoValue_SpannerWriteSchemaTransformProvider_SpannerWriteSchemaTransformConfiguration
           .Builder();
     }
 
     @AutoValue.Builder
     public abstract static class Builder {
+      public abstract Builder setProjectId(String projectId);
+
       public abstract Builder setInstanceId(String instanceId);
 
       public abstract Builder setDatabaseId(String databaseId);
 
       public abstract Builder setTableId(String tableId);
 
+      public abstract Builder setErrorHandling(ErrorHandling errorHandling);
+
       public abstract SpannerWriteSchemaTransformConfiguration build();
     }
+
+    public void validate() {
+      String invalidConfigMessage = "Invalid Spanner Write configuration: ";
+
+      checkArgument(
+          !Strings.isNullOrEmpty(this.getProjectId()),
+          invalidConfigMessage + "Project ID for a Spanner Write must be specified.");
```

Comment:
ProjectId is not required - probably why the IT is failing to build transform since it does not pass project
```suggestion
```